lightning_server.cc

```diff
@@ -7,7 +7,11 @@
 #include "response.h"
 
 #include <iostream>
+#include <boost/thread.hpp>
 #include <cstddef>
+#include <boost/smart_ptr.hpp>
+
+typedef boost::shared_ptr<tcp::socket> socket_ptr;
```

Comment:
Does this have to be a `shared_ptr`? Also, IIRC, Boost's shared_ptr comes from pre-C++11 days, when we did not yet have unique_ptr.